src/Fields/fieldvector.jl

```diff
@@ -332,24 +322,41 @@ end
 
 @inline function Base.copyto!(
     dest::FieldVector,
-    bc::Base.Broadcast.Broadcasted{FieldVectorStyle},
+    bc::Union{FieldVector, Base.Broadcast.Broadcasted{FieldVectorStyle}},
 )
     map(propertynames(dest)) do symb
         Base.@_inline_meta
-        p = parent(getfield(_values(dest), symb))
-        copyto!(p, transform_broadcasted(bc, symb, axes(p)))
+        array = parent(getfield(_values(dest), symb))
+        bct = transform_broadcasted(bc, symb, axes(array))
+        copyto_per_field!(array, bct, DataLayouts.device_dispatch(array))
+    end
+    return dest
+end
+
+@inline function Base.copyto!(
+    dest::FieldVector,
+    bc::Base.Broadcast.Broadcasted{Style},
+) where {
+    Style <:
+    Union{Base.Broadcast.AbstractArrayStyle{0}, Base.Broadcast.Style{Tuple}},
+}
+    map(propertynames(dest)) do symb
+        Base.@_inline_meta
```

Comment:
Is this inlining the anonymous function for the map?

Reply:
Yes